### PR TITLE
Localize and fix coloring of combo defaults

### PR DESCRIPTION
**Pull request by Hdt80bro**
* Adds localization for the " (default)" label for combo boxes. I should have only added localization support, but I tried to add the database keys based on other translation work available.
* Fixes the coloring of the title not being yellow when the default item is immediately selected
* Moves the ad-hoc callbacks to the combo class
* Other refactoring of the file
* Adds some annotation stuff in the surrounding area

### CODE REVIEW

File: lua/ui/controls/combo.lua [hunk 1]
```diff
@@ -508,92 +644,37 @@ BitmapCombo = Class(Group) {
...
-                    local sound = Sound({Cue = self.mClickCue, Bank = "Interface",})
-                    PlaySound(sound)
-                end
-            end
-            self:OnEvent(event)
-
-            return eventHandled
-        end
-
-        self.OnDisable = function(self)
```
Comment (Garanas) [inline lua/ui/controls/combo.lua:568]: Interesting how these were added manually, for each instance 😄

**Review by Garanas — APPROVED** [review]
works in-game, can't produce any errors

**Comment by Garanas** [comment]
@Hdt80bro 

This changes to the combobox are apparently not entirely logically equivalent, as it appears this breaks the minimap functionality where you can switch to the spot. I'm talking about these:

image

Don't ask why it uses a combo box, I don't know. But can you look into this?

**Comment by Hdt80bro** [comment]
Uh, sure

**Comment by Hdt80bro** [comment]
The logical difference comes from closing the methods with the `self` upvalue from the initializer instead of using the called method control. This revealed a latent bug [here](https://github.com/FAForever/fa/blob/deploy/fafdevelop/lua/ui/lobby/lobby.lua#L4812-L4814) where the event handler was not being called on the right control.

**Pull request merged**